src/include/sof/lib_manager.h

```diff
@@ -0,0 +1,204 @@
+ *   |                       |                            | -----------------------> |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |  Verify Manifest         |         |
+ *   |                       |                            | -------------------------|-------> |
+ *   |                       |                            |  results                 |         |
+ *   |                       |                            | <------------------------|-------- |
+ *   |                       |                            |   (IF AUTH_API_ENABLED)  |         |
+ *   |                       |                            |                          |         |
+ *   |                       |                            | Update Library           |         |
+ *   |                       |                            | descriptors table        |         |
+ *   |                       |                            | -------                  |         |
+ *   |                       |                            |        |                 |         |
+ *   |                       |                            | <------                  |         |
+ *   |                       |       return status        |                          |         |
+ *   |                       | <------------------------- |                          |         |
+ *   | Complete IPC request  |                            |                          |         |
+ *   | <-------------------  |                            |                          |         |
+ *
+ * Driver                   IPC4                       Library                     MEMORY    ENTITY
+ *                         Handler                     Manager                               VERIF.
+ */
+
+#ifndef __SOF_LIB_MANAGER_H__
+#define __SOF_LIB_MANAGER_H__
+
+#include <stdint.h>
+#include <rimage/cavs/cavs_ext_manifest.h>
+#include <rimage/sof/user/manifest.h>
+
+#define LIB_MANAGER_MAX_LIBS				16
+#define LIB_MANAGER_LIB_ID_SHIFT			12
+#define LIB_MANAGER_LIB_NOTIX_MAX_COUNT		4
+
+#define LIB_MANAGER_GET_LIB_ID(module_id) ((module_id) >> LIB_MANAGER_LIB_ID_SHIFT)
+#define LIB_MANAGER_GET_MODULE_INDEX(module_id) ((module_id) & 0xFFF)
+
+#ifdef CONFIG_LIBRARY_MANAGER
+struct ipc_lib_msg {
+	struct ipc_msg *msg;
+	struct list_item list;
+};
+
+static SHARED_DATA struct ext_library {
+	struct k_spinlock lock;	/* locking mechanism */
+	struct sof_man_fw_desc *desc[LIB_MANAGER_MAX_LIBS];
+	uint32_t mods_exec_load_cnt;
+	struct ipc_lib_msg *lib_notif_pool;
+	uint32_t lib_notif_count;
+};
+
+/* lib manager context, used by lib_notification */
+extern struct tr_ctx lib_manager_tr;
+
+static inline struct ext_library *ext_lib_get(void)
+{
+	return sof_get()->ext_library;
```

Comment:
error: 'struct sof' has no member named 'ext_library'

Reply:
In previous your comment you pointed to struct ext_library defined in this include.

Reply:
First to make the PR compile:
```diff
diff --git a/app/overlays/tgl/ipc4_overlay.conf b/app/overlays/tgl/ipc4_overlay.conf
index b986e772f5f9..ca0086981921 100644
--- a/app/overlays/tgl/ipc4_overlay.conf
+++ b/app/overlays/tgl/ipc4_overlay.conf
@@ -11,3 +11,5 @@ CONFIG_PCM_CONVERTER_FORMAT_S16_C32_AND_S32_C32=y
 CONFIG_PCM_CONVERTER_FORMAT_S16_C32_AND_S24_C32=y
 CONFIG_PCM_CONVERTER_FORMAT_S16_C32_AND_S16_C32=y
 
+CONFIG_LIBRARY_MANAGER=y
+
diff --git a/src/Kconfig b/src/Kconfig
index a24d0fc402cb..a98564f16702 100644
--- a/src/Kconfig
+++ b/src/Kconfig
@@ -19,3 +19,5 @@ rsource "schedule/Kconfig"
 rsource "ipc/Kconfig"
 
 rsource "math/Kconfig"
+
+rsource "library_manager/Kconfig"
```
then compilation will fail:
```
sof/zephyr/../src/include/sof/lib_manager.h:88:1: warning: useless storage class specifier in empty declaration
   88 | };
      | ^
sof/zephyr/../src/include/sof/lib_manager.h: In function 'ext_lib_get':
sof/zephyr/../src/include/sof/lib_manager.h:95:18: error: 'struct sof' has no member named 'ext_library'
   95 |  return sof_get()->ext_library;
      |                  ^~
```

Reply:
`struct sof` via `src/include/sof/sof.h` does not have a member named `ext_library`

Reply:
This feature won't work at tgl. It is mtl and newer only. Additionally that feature needs zephyr native drivers, dma, ipc4. I'm not sure what are you aiming for. This feature is works since the beggining on developer integration branch.

Reply:
@ujfalusi correct, there is lack of struct pointer. Fixed.